core/formats/formats_10.cpp

```diff
@@ -296,24 +293,17 @@ struct pay_buffer : skip_buffer {
   uint32_t last{};           // last start offset
 };                           // pay_buffer
 
-// Buffer carrying competitive block scores
-class skip_score_stats {
- public:
-  static void skip(index_input& in) { in.read_vint(); }
-
-  static uint32_t read(index_input& in) { return in.read_vint(); }
-
-  void add(uint32_t value) noexcept { freq_ = std::max(value, freq_); }
-
-  void add(skip_score_stats rhs) noexcept { add(rhs.freq_); }
-
-  void reset() noexcept { freq_ = 0; }
+std::vector<irs::WandWriter::ptr> PrepareWandWriters(ScorersView scorers,
+                                                     size_t max_levels) {
+  scorers = scorers.subspan(0, kMaxScorers);
```

Comment:
`subspan` does not check the limits. So if `scorers` size less than `kMaxScorers` this would break. Suggest to just limit the number of writers. See below.